docs/_posts/2015-09-30-react-render-and-top-level-api.md

```diff
@@ -0,0 +1,133 @@
+```
+
+If you're integrating with an object oriented API around you this might be confusing. For example, if you're using React to build a Web Component (Custom Element). In object oriented programming, all state lives on each object instance and you apply changes incrementally by mutating that state, one at a time. Not all at once like React does.
+
+We used to have a helper function called `setProps` which allowed you to update only a few properties at a time. Unfortunately this API lived on a component instance, required React to keep this state internally and wasn't very natural anyway. Therefore, we're deprecating it and suggest that you build it into your own wrapper instead.
+
+Here's some boilerplate to get you started. It is a 0.14 migration path for codebases using `setProps` and `replaceProps`.
+
+```js
+class ReactComponentRenderer {
+  constructor(klass, container) {
+    this.klass = klass;
+    this.container = container;
+    this.props = {};
+    this.component = null;
+  }
+
+  replaceProps(props, callback) {
+    this.props = {};
+    this.setProps(props, callback);
+  }
+
+  setProps(partialProps, callback) {
+    if (this.klass == null) {
+      console.warn(
+        'setProps(...): Can only update a mounted or ' +
+        'mounting component. This usually means you called setProps() on ' +
+        'an unmounted component. This is a no-op.'
+      );
+      return;
+    }
+    Object.assign(this.props, partialProps);
+    var element = React.createElement(this.klass, this.props);
+    this.component = ReactDOM.render(element, this.container, callback);
+  }
+
+  unmount() {
+    ReactDOM.unmountComponentAtNode(this.container);
+    this.klass = null;
+  }
+}
+```
+
+Object-oriented APIs don't look like that though. They use setters and methods. I think we can do better. If you know more about the component API that you're rendering, you can create a more natural object-oriented API around your React component.
+
+```js
+class ReactVideoPlayer {
+  constructor(url, container) {
+    this._container = container;
+    this._url = url;
+    this._isPlaying = false;
+    this._render();
+  }
+
+  _render() {
+    ReactDOM.render(
+      <VideoPlayer url={this._url} playing={this._isPlaying} />,
+      this._container
+    );
+  }
+
+  get url() {
+    return this._url;
+  }
+
+  set url(value) {
+    this._url = value;
+    this._render();
+  }
+
+  play() {
+    this._isPlaying = true;
+    this._render();
+  }
+
+  pause() {
+    this._isPlaying = false;
+    this._render();
+  }
+
+  destroy() {
+    ReactDOM.unmountComponentAtNode(this._container);
+  }
+}
+```
+
+It is our hope that this implementation will showcase a pattern whereby components can transcend the divide between a declarative and imperative APIs. This example shows how to provide an imperative API on top of a declarative one. Similarly, the reverse can be done, and a declarative wrapper can be used when exposing a WebComponent as a React component.
```

Comment:
It's a little academic/philosophical. It doesn't add anything. We can just remove it.

The last sentence is a bit of a tease since this article doesn't say anything about that part. It's just frustrating. Seems like it's better to remove that and use it as the lead in for a follow up article.